test/conftest.py

```diff
@@ -354,6 +354,23 @@ def embed_documents(self, documents):
         return np.random.rand(len(documents), self.embedding_dim)
 
 
+class MockFilterRetriever(MockRetriever, FilterRetriever):
```

Comment:
I think you don't need to actually inherit from `FilterRetriever`

Reply:
Yeah, gotcha. Now it's generic, inheriting from BaseRetriever and the name has changed too